internal/server/data/data_test.go

```diff
@@ -213,3 +214,37 @@ func TestDB_Begin(t *testing.T) {
 		})
 	})
 }
+
+func TestLongRunningQueriesAreCancelled(t *testing.T) {
+	runDBTests(t, func(t *testing.T, db *DB) {
+
+		t.Run("Gorm", func(t *testing.T) {
+			started := time.Now()
+
+			ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+			defer cancel()
+			tx := db.WithContext(ctx)
+			err := tx.Exec("select pg_sleep(10);").Error
+			assert.Error(t, err, "timeout: context deadline exceeded")
+
+			elapsed := time.Since(started)
+			assert.Assert(t, elapsed < 3*time.Second, "query should have timed out and been cancelled")
```

Comment:
This assertion can be stricter if you use `assert.DeepEqual` with `DurationWithThreshold`. Not blocking since this is likely sufficient